sdk/core/azure-core/src/main/java/com/azure/android/core/credential/TokenRequestObservableAuthInterceptor.java

```diff
@@ -28,30 +28,31 @@
     private final ReentrantLock sendRequestLock = new ReentrantLock();
 
     /**
-     * Creates TokenRequestObservableAuthInterceptor.
+     * Creates a {@link TokenRequestObservableAuthInterceptor}.
      *
-     * @param scopes the scope for the requested token
+     * @param scopes The scope for the requested token.
      */
     public TokenRequestObservableAuthInterceptor(List<String> scopes) {
         this.scopes = new ArrayList<>(scopes);
     }
 
     /**
-     * @return the token request observable that UI can Observe for access-token
-     * request coming from this interceptor.
+     * @return The {@link TokenRequestObservable} that the UI can observe for an {@link AccessToken} request coming
+     * from this interceptor.
      */
     public TokenRequestObservable getTokenRequestObservable() {
         return this.requestObservable;
     }
 
+    @NonNull
     @Override
-    public Response intercept(Chain chain) throws IOException {
-        if (this.accessToken != null && !this.accessToken.isExpired()) {
-            return setAuthenticationHeader(chain, this.accessToken);
-        } else {
+    public Response intercept(@NonNull Chain chain) throws IOException {
+        if (this.accessToken == null || this.accessToken.isExpired()) {
```

Comment:
I know you didn't actually change this code, but in iOS we don't check if the token is expired, we just let the service call fail with the appropriate error response. This goes with the pattern of not implementing logic in the client library, that is also implemented on the service side (since you can never guarantee the same logic will be implemented identically in both places).

Can or should we remove this check? If the answer is yes, it doesn't need to be done in this PR, we should just file an issue.

Reply:
Checking for expiration is simple enough with this `AccessToken` class and I believe that we should not make additional calls if we can easily know if the token is still valid, however, I do not know what other checks might be ocurring in the service side, so discussing this a little further sounds like a good idea.